doc/source/cluster/kubernetes/examples/rayserve-llm-example.md

```diff
@@ -0,0 +1,164 @@
+(kuberay-rayservice-llm-example)=
+
+# Serve a Large Language Model using Ray Serve LLM on Kubernetes
+
+This guide provides a step-by-step walkthrough for deploying a Large Language Model (LLM) using Ray Serve LLM on Kubernetes. Leveraging KubeRay, Ray Serve, and vLLM, this guide deploys the  `Qwen/Qwen2.5-7B-Instruct` model from HuggingFace, enabling scalable, efficient, and OpenAI-compatible LLM serving within a Kubernetes environment. For more information on Ray-Serve LLM, visit [here](https://docs.ray.io/en/latest/serve/llm/serving-llms.html).
+
+## Step 1: Create a Kubernetes cluster with GPUs
+
+Follow aws-eks-gpu-cluster.md or gcp-gke-gpu-cluster.md to create a Kubernetes cluster.
+
+### Prerequisites
+In this example, model weights are downloaded from the [Qwen/Qwen2.5-7B-Instruct](https://huggingface.co/Qwen/Qwen2.5-7B-Instruct) Hugging Face repository. In order to properly finish this guide, you must fulfill the following requirements:
+* A [Hugging Face Account](https://huggingface.co/) and a Hugging Face [Access Token](https://huggingface.co/settings/tokens) with read access to gated repos.
+* In your Ray Serve configuration, set the `HUGGING_FACE_HUB_TOKEN` environment variable to this token to enable model downloads.
+* Any supported GPU. You can refer [here](https://docs.ray.io/en/latest/cluster/kubernetes/user-guides/k8s-cluster-setup.html#kuberay-k8s-setup) for more information
+
+## Step 2: Install the KubeRay Operator
+
+Install the most recent stable KubeRay operator from the Helm repository by following [Deploy a KubeRay operator](kuberay-operator-deploy). The KubeRay operator Pod must be on the CPU node if you set up the taint for the GPU node pool correctly.
+
+## Step 3: Create a Kubernetes Secret containing your Hugging Face access token
```

Comment:
The step should be:

* Download the YAML.
* Update the Secret in the YAML
* Apply the YAML